### ISSUE TITLE
wrong documentation for cds.foreach

### ISSUE THREAD

**Issue by kulliax** [cap-js/docs#1499]
The documentation describes a behaviour that is not implemented:
https://github.com/cap-js/docs/blob/44749e0d6f13065007ee43f94d9ce5ccad70205d/node.js/core-services.md?plain=1#L1064

comment in source:

>   node_modules/@sap/cds/lib/srv/srv-api.js
>   /**
>    * Streaming API variant of .run(). Subclasses should override this to support real streaming.
>    * The default implementation doesn't stream, but simply invokes the callback on each row.
>    * The callback function is invoked with (row, index).
>    */

Does this mean that we have to implement this ourselves? I can't find a subclass for this or is that nearly planned? We've come across this point several times and after the big release of the revised documentary we wanted to address it here as well.

For example, the hana with the stream module actually supports this function.

**Comment by renejeglinsky** [comment]
Hi @kulliax ,
thanks for opening this issue and sorry for the late response.

@sjvans Could you check this feedback if that is not implemented or if we didn't describe it well enough? Thanks.

**Comment by sjvans** [comment]
correct, it is not yet implemented by any subclass (cds.DatabaseService, mainly). i'll try to clarify that. the implementation in `cds.Service` itself is there to ensure the API and will/ must always remain a mock.

https://cap.cloud.sap/docs/node.js/core-services#how-to-provide-custom-service-implementations documents how to provide custom service implementations. the `BooksService` in that example could have a foreach function that overrides the non-streaming default impl provided by CAP.

**Comment by kulliax** [comment]
Hi @sjvans as I heard in one of the last roundtables, you have it on the list. Is there a roadmap for it somewhere?

**Comment by sjvans** [comment]
hi @kulliax 

it's in the backlog (sorry, sap internal), but not yet scheduled.

best,
sebastian
